src/lib/uri/uri.c

```diff
@@ -170,6 +179,35 @@ uri_create(struct uri *uri, const char *str)
 	return 0;
 }
 
+static int
+uri_format_param(char **str, int *len, const struct uri_param *param)
+{
+	int total = 0;
+	if (param->value_count == 0)
+		SNPRINT(total, snprintf, *str, *len, "%s", param->name);
```

Comment:
Please, make a return here when value_count was 0. The loop below is confusing otherwise. And useless anyway.

Reply:
fixed